src/Build/BuildCheck/Acquisition/BuildCheckAcquisitionModule.cs

```diff
@@ -4,19 +4,74 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using Microsoft.Build.BuildCheck.Analyzers;
+using System.Reflection;
+using Microsoft.Build.BackEnd.Logging;
 using Microsoft.Build.BuildCheck.Infrastructure;
+using Microsoft.Build.Experimental.BuildCheck;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
 
-namespace Microsoft.Build.BuildCheck.Acquisition;
-
-internal class BuildCheckAcquisitionModule
+namespace Microsoft.Build.BuildCheck.Acquisition
 {
-    private static T Construct<T>() where T : new() => new();
-    public BuildAnalyzerFactory CreateBuildAnalyzerFactory(AnalyzerAcquisitionData analyzerAcquisitionData)
+    internal class BuildCheckAcquisitionModule : IBuildCheckAcquisitionModule
     {
-        // Acquisition module - https://github.com/dotnet/msbuild/issues/9633
-        return Construct<SharedOutputPathAnalyzer>;
+        private readonly ILoggingService _loggingService;
+
+        internal BuildCheckAcquisitionModule(ILoggingService loggingService)
+        {
+            _loggingService = loggingService;
+        }
+
+#if FEATURE_ASSEMBLYLOADCONTEXT
+        /// <summary>
+        /// AssemblyContextLoader used to load DLLs outside of msbuild.exe directory
+        /// </summary>
+        private static readonly CoreClrAssemblyLoader s_coreClrAssemblyLoader = new();
+#endif
+
+        public IEnumerable<BuildAnalyzerFactory> CreateBuildAnalyzerFactories(AnalyzerAcquisitionData analyzerAcquisitionData, BuildEventContext buildEventContext)
+        {
+            try
+            {
+                Assembly? assembly = null;
+#if FEATURE_ASSEMBLYLOADCONTEXT
+                assembly = s_coreClrAssemblyLoader.LoadFromPath(analyzerAcquisitionData.AssemblyPath);
+#else
+                assembly = Assembly.LoadFrom(analyzerAcquisitionData.AssemblyPath);
+#endif
+
+                IEnumerable<Type> analyzerTypes = assembly.GetTypes().Where(t => typeof(BuildAnalyzer).IsAssignableFrom(t));
+
+                if (analyzerTypes.Any())
+                {
+                    var analyzersFactory = new List<BuildAnalyzerFactory>();
```

Comment:
NIT: should be plural

```suggestion
                    var analyzersFactories = new List<BuildAnalyzerFactory>();
```